.github/workflows/_shared-first-time-setup.yml

```
@@ -0,0 +1,65 @@
+name: "Repository First Time Setup automation (shared)"
+
+on:
+  workflow_call:
+    inputs:
+        esf_version:
+          type: string
+          default: '7.2.2'
+          required: true
+        archetype_url:
+          type: string
+          default: 'https://changeme'
+          required: true
+        groupId:
+          type: string
+          default: ''
+          required: true
+        artifactId:
+          type: string
+          default: ''
+          required: true
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Download ESF archetype
```

The repository checkout step is missing:

```suggestion
    steps:
    
     - name: Checkout ${{ github.ref }} branch in ${{ github.repository }} repository.
      uses: actions/checkout@v3

    - name: Download ESF archetype
```